Mathlib/Analysis/Calculus/MeanValue.lean

```diff
@@ -747,6 +747,126 @@ theorem exists_deriv_eq_slope : ∃ c ∈ Ioo a b, deriv f c = (f b - f a) / (b
   exists_hasDerivAt_eq_slope f (deriv f) hab hfc fun x hx =>
     ((hfd x hx).differentiableAt <| IsOpen.mem_nhds isOpen_Ioo hx).hasDerivAt
 
+/-- Lagrange's **Mean Value Theorem**, `deriv` version. -/
+theorem exists_deriv_eq_slope' : ∃ c ∈ Ioo a b, deriv f c = slope f a b := by
+  rw [slope_def_field]
+  exact exists_deriv_eq_slope f hab hfc hfd
+
+/-- A real function whose derivative tends to infinity from the right at a point is not
+differentiable on the right at that point -/
+theorem not_differentiableWithinAt_of_deriv_tendsto_atTop_Ioi (f : ℝ → ℝ) {a : ℝ}
+    (hf : Tendsto (deriv f) (𝓝[>] a) atTop) : ¬ DifferentiableWithinAt ℝ f (Ioi a) a := by
```

Comment:
Does this generalize to `nhdsWithin a s` somehow?

Reply:
I'm not too sure how to do it in that case -- this uses the mean value theorem, which won't work if there are "holes" in `s`.

Reply:
@eric-wieser I also don't understand what lemma statement you could have in mind...

Reply:
The statement I had in mind is
```suggestion
    (hf : Tendsto (deriv f) (nhdsWithin a s) atTop) : ¬ DifferentiableWithinAt ℝ f s a := by
```
perhaps also with `IsOpen s`; though I don't make any particularly strong claim that this is reasonable. My comment was for another more-qualified reviewer to remark the sensibility of, not a direct request for a change.

Reply:
Not sensible. This requires MVT.